plugins/auth-oauth2/src/grants/implicit.ts

```diff
@@ -42,25 +42,33 @@ export function getImplicit(
     }
 
     const authorizationUrlStr = authorizationUrl.toString();
+    let foundAccessToken = false;
     let { close } = await ctx.window.openUrl({
       url: authorizationUrlStr,
       label: 'oauth-authorization-url',
+      async onClose() {
+        if (!foundAccessToken) {
+          reject(new Error('Authorization window closed'));
+        }
+      },
       async onNavigate({ url: urlStr }) {
         const url = new URL(urlStr);
         if (url.searchParams.has('error')) {
           return reject(Error(`Failed to authorize: ${url.searchParams.get('error')}`));
         }
 
-        // Close the window here, because we don't need it anymore
-        close();
-
         const hash = url.hash.slice(1);
         const params = new URLSearchParams(hash);
-        const idToken = params.get('id_token');
-        if (idToken) {
-          params.set('access_token', idToken);
-          params.delete('id_token');
```

Comment:
Not sure why this was here in the first place but it (somewhat obviously) breaks stuff. Getting the token works fine at first glance but then auth to my API doesn't work because it doesn't receive the right token.

Reply:
I never had an implicit OAuth API to test with so it was just wrong 🙈

Reply:
FWIW I tested it with Cognito and a free Auth0 account, in case you're looking for something to test with in the future!